### PR TITLE
KUZ-410: normalize chaining

### PR DESCRIPTION
**Pull request by scottinet**
* Removed chaining capabilities from functions with required callbacks
* Added chaining capabilities to functions with optional callbacks

### CODE REVIEW

File: source/includes/_kuzzleObject.md [hunk 1]
```diff
@@ -1843,10 +1815,6 @@ If the login attempt fails, the `loginAttempt` event is fired with the following
 By default, kuzzle embed the plugin [kuzzle-plugin-auth-passport-local](https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local) which provide `local` stored password authentication strategy.
 This strategy require `username` and `password` as `credentials`
 
-### Return value
```
Comment (ballinette) [inline source/includes/_kuzzleObject.md]: Did you forgot to remove the chaining also for `logout` (next section below), or is it still necessary ?

Comment (scottinet) [inline source/includes/_kuzzleObject.md]: Chaining has been kept intentionally.
I think we should not prevent code like this one:

```js
kuzzle
   .logout()
   .disconnect();
```

Comment (ballinette) [inline source/includes/_kuzzleObject.md]: OK

**Comment by ballinette** [comment]
+1

**Pull request merged**